embed_grader/serapis/views/courses.py

```diff
@@ -169,6 +171,88 @@ def unenroll_course(request, course_id):
     }
     return render(request, 'serapis/unenroll_course.html', template_context)
 
+@login_required(login_url='/login/')
+def download_csv(request, course_id):
+    user = User.objects.get(username=request.user)
+    course = Course.objects.get(id=course_id)
+    print(course)
+    students = []
+    headers = ['Last Name', 'First Name', 'UID']
+
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = 'attachment; filename="%s_scores.csv"' % course
+    writer = csv.writer(response)
+
+    if not user.has_perm('download_csv', course):
+        return HttpResponse("Not enough privilege.")
+
+    assignment_list = Assignment.objects.filter(course_fk=course_id).order_by('-id')
+    for h in assignment_list:
+        headers.append(h.name)
+
+    writer.writerow(headers)
+
+    assignment_ids = []
+
+    for assign in assignment_list:
+        assignment_ids.append(assign.id)
+
+    print(assignment_ids)
+
+    students = [o.user_fk for o in CourseUserList.objects.filter(course_fk=course)]
+    scores = {}
```

Comment:
Let's put a comment to briefly summarize the structure of `scores`: 

```
scores[student][a_idx] means number of points the student got in (a_idx)-th assignment, and 
a_idx refers to the index of assignment_list.
```

BTW in this repo, we keep no more than 100 characters each line in python code